rust-runtime/aws-smithy-runtime/src/client/orchestrator/auth.rs

```diff
@@ -3,16 +3,17 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-use super::phase::Phase;
+use crate::client::orchestrator::AttemptCheckpoint;
+use crate::{bail, handle_err};
 use aws_smithy_http::result::SdkError;
 use aws_smithy_runtime_api::client::interceptors::context::Error;
 use aws_smithy_runtime_api::client::orchestrator::{BoxError, ConfigBagAccessors, HttpResponse};
 use aws_smithy_runtime_api::config_bag::ConfigBag;
 
 pub(super) async fn orchestrate_auth(
-    dispatch_phase: Phase,
+    mut checkpoint: AttemptCheckpoint,
     cfg: &ConfigBag,
-) -> Result<Phase, SdkError<Error, HttpResponse>> {
+) -> Result<AttemptCheckpoint, SdkError<Error, HttpResponse>> {
```

Comment:
Instead of checkpoints, what do you think of referring to these phases as part of the `OrchestratorContext`? If we use the `Context` verbiage, then people will naturally associate it with `InterceptorContext` and that's good because the two are related.

Reply:
Going to leave it as is for now. We can continue to think about the name.